### ISSUE TITLE
[Bug] Tab freeze when buying stuff on allegro.pl

### ISSUE THREAD

**Issue by Mikaka27** [mozilla-mobile/fenix#16234]
Unfortunately I can only reproduce this on allegro.pl site (which is polish equivalent of ebay.com). You need a real account there to reproduce, hopefully someone from Poland is on the team.
I don't think it's tied to this specific website, rather this website is doing something which is triggering this.

## Steps to reproduce
1. Open firefox
2. Go to allegro.pl
3. Login
4. Search for something to buy
5. Click buy now
6. Select "paczkomat" delivery type, and change delivery point (optional, sometimes it freezes without this step)

### Expected behavior
I expect to be able to finish the purchase. This includes selecting the delivery point from the map.
Strangely enough this issue is not present in android emulator, hence here's the link to video with "expected" behavior.
https://www.dropbox.com/s/75ahzmg32lc3qtg/ComputerFirefoxBug.mp4?dl=0

You can see that at the end you can see the map from which I choose delivery point.
You can see pixelated parts in the videos, I did this to hide my real account data, please ignore this.

### Actual behavior
Actual behavior is that the tab freezes completely, you cannot click anything. Instead of the map to choose delivery point you see white screen only.
I'm adding a video with "actual" behavior here:
https://www.dropbox.com/s/zhytrt3yk8arkic/PhoneFirefoxBug.mp4?dl=0

What you see in this video is that I do exactly the same thing as in Android emulator, but I get a different result.
When tab freezes I do the following:
1. Wait some time to see if it fixes itself
2. Close tab from browser UI
3. Check if process is killed (it is not, it continues to use 100% cpu)
4. Try to open allegro.pl website in new tab again
5. This fails, only blank page is displayed (besides that there's no indication that something is wrong in firefox)
6. I kill the "tab0" process with kill signal from terminal emulator (terminating firefox from android settings also works)
7. Now I can open allegro.pl in new tab successfully

I can reproduce this 100% of the time on 2 different devices.

### Device information

* Android device: Samsung Galaxy J7 2017 and Xiaomi Redmi 4A
* Fenix version: 82.1.1 and todays nightly however I saw this behavior before
 
### Additional information
I tried to debug this, here are things I've tried:
1. Connect android phone via usb and hit "pause" in debugger section to see if page is stuck in javascript loop. It is not, pause button does nothing
2. Use performance profiler to see what is going on, this also fails retrieval of result is timing out, and silently failing, I get no results
3. Debug fenix repository code on my real android phone, this also fails application is crashing right after it says "Waiting for debugger"

I'm out of ideas for now :(

Please comment if something needs further explanation.
Best regards

**Comment by kbrosnan** [comment]
I would try getting a [Firefox Profile of the behavior](https://profiler.firefox.com/docs/#/./guide-remote-profiling)

**Comment by Mikaka27** [comment]
Hi, I've managed to attach debugger to real device.
Page gets stuck in the same way. When it does even though I'm scrolled halfway when I try to scroll up the "pull to refresh" icon appears as if it cannot go back up.

I didn't get anything useful out of this yet.
I got a thread dump from locked website (I hit pause in debugger).

I think I would need to have symbols of gecko and see there.

[threads_report.txt](https://github.com/mozilla-mobile/fenix/files/5454507/threads_report.txt)
